frontend/src/metabase/admin/datamodel/components/database/MetadataTableList.jsx

```diff
@@ -37,44 +52,33 @@ export default class MetadataTableList extends Component {
     });
   }
 
+  renderTable = table => (
+    <li key={table.id}>
+      <a
+        className={cx(
+          "AdminList-item flex align-center no-decoration text-wrap",
+          {
+            selected: this.props.tableId === table.id,
+          },
+        )}
+        onClick={this.props.selectTable.bind(null, table)}
+      >
+        {table.display_name}
+      </a>
+    </li>
+  );
+
   render() {
     let queryableTablesHeader, hiddenTablesHeader;
-    const queryableTables = [];
-    const hiddenTables = [];
-
-    if (this.props.tables) {
-      const tables = _.sortBy(this.props.tables, "display_name");
-      _.each(tables, table => {
-        const selected = this.props.tableId === table.id;
-        const row = (
-          <li key={table.id}>
-            <a
-              className={cx(
-                "AdminList-item flex align-center no-decoration text-wrap",
-                {
-                  selected,
-                },
-              )}
-              onClick={this.props.selectTable.bind(null, table)}
-            >
-              {table.display_name}
-            </a>
-          </li>
-        );
-        const regex = this.state.searchRegex;
-        if (
-          !regex ||
-          regex.test(table.display_name) ||
-          regex.test(table.name)
-        ) {
-          if (table.visibility_type) {
-            hiddenTables.push(row);
-          } else {
-            queryableTables.push(row);
-          }
-        }
-      });
-    }
+    const regex = this.state.searchRegex;
+    const [hiddenTables, queryableTables] = _.chain(this.props.tables)
```

Comment:
Feels like the component shouldn't have to necessarily do this much work inside of `render` to get the hidden and queryable tables. Might be a candidate for moving to `mapStateToProps`

Reply:
The filter step in this chain relies on component state, but I extracted it to a separate component function rather than including it in the body of render.